From 1dbdef4043bbb8a818888aa09e5dd232ff0332a5 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Markus=20M=C3=BCtzel?= <markus.muetzel@gmx.de>
Date: Tue, 12 Sep 2023 10:29:59 +0200
Subject: [PATCH] qt5-static: Fix building in 32 bit environment

Adopt patch from ec6dda0bd483fb45d37f2438723913eee7e9b803

diff --git a/mingw-w64-qt5-static/0005-qtconnectivity-add-callback-attribute.patch b/mingw-w64-qt5-static/0005-qtconnectivity-add-callback-attribute.patch
new file mode 100644
index 000000000..db4612ba1
--- /dev/null
+++ b/mingw-w64-qt5-static/0005-qtconnectivity-add-callback-attribute.patch
@@ -0,0 +1,16 @@
+--- a/qtconnectivity/src/bluetooth/qbluetoothservicediscoveryagent_win.cpp
++++ b/qtconnectivity/src/bluetooth/qbluetoothservicediscoveryagent_win.cpp
+@@ -206,12 +206,7 @@
+     return sequence;
+ }
+
+-#if defined(Q_CC_MINGW)
+-# define SDP_CALLBACK
+-#else
+-# define SDP_CALLBACK QT_WIN_CALLBACK
+-#endif
+-static BOOL SDP_CALLBACK bluetoothSdpCallback(ULONG attributeId, LPBYTE valueStream, ULONG streamSize, LPVOID param)
++static BOOL QT_WIN_CALLBACK bluetoothSdpCallback(ULONG attributeId, LPBYTE valueStream, ULONG streamSize, LPVOID param)
+ {
+     QBluetoothServiceInfo *result = static_cast<QBluetoothServiceInfo*>(param);
+ 
diff --git a/mingw-w64-qt5-static/PKGBUILD b/mingw-w64-qt5-static/PKGBUILD
index fe0904081..aa79ff423 100644
--- a/mingw-w64-qt5-static/PKGBUILD
+++ b/mingw-w64-qt5-static/PKGBUILD
@@ -12,9 +12,9 @@ _realname=qt5-static
 pkgbase="mingw-w64-${_realname}"
 pkgname="${MINGW_PACKAGE_PREFIX}-${_realname}"
 pkgver=5.15.10
-pkgrel=1
+pkgrel=2
 arch=('any')
-mingw_arch=('mingw64' 'ucrt64' 'clang64' 'clangarm64')
+mingw_arch=('mingw32' 'mingw64' 'ucrt64' 'clang64' 'clang32' 'clangarm64')
 pkgdesc="A cross-platform application and UI framework (static) (mingw-w64)"
 url='https://www.qt.io/'
 license=('spdx:GPL-2.0-only' 'spdx:GPL-3.0-or-later' 'spdx:LGPL-2.1-or-later' 'spdx:LGPL-3.0-only' 'spdx:Qt-GPL-exception-1.0' 'spdx:GFDL-1.3-or-later')
@@ -43,6 +43,7 @@ source=(https://download.qt.io/official_releases/qt/${pkgver%.*}/${pkgver}/singl
         0002-qt-5.8.0-configure-gcc-before-clang.patch
         0003-enable-avx-support.patch
         0004-fix-linking-again-different-static-libs.patch
+        0005-qtconnectivity-add-callback-attribute.patch
         0006-qt-5.3.0-win_flex-replace.patch
         0007-qt-5.3.0-win32-g-Enable-static-builds.patch
         0008-qt-5.3.0-win32-g-Add-QMAKE_EXTENSION_IMPORTLIB-defaulting-to-.patch
@@ -80,6 +81,7 @@ sha256sums=('b545cb83c60934adc9a6bbd27e2af79e5013de77d46f5b9f5bb2a3c762bf55ca'
             '27ce2161b5dbea2fcce6b15948ab722a71036d056420854f6554969f683583bb'
             '15c9fdf4d54628e64a9b7e456a5308aecf0f1c8e33a5aff0ab0bcac0300a9568'
             'e04033364f5ad77906b4100c34572df57bd793a55bc33b7d8a753a2cc60af259'
+            '3be09af4b54ee0792a9e91fe7a1cf07d10f65f68564b93c86684599c31530c91'
             '18fd2fa42215ac47b3b314261ab98cbe65f8231429e4f29a152288a3ca93daf1'
             '4e154fbc9059a096c351d019da6b18c907b1d8b06e028f48c7365f62bcd0edc9'
             'e2ec5e67bdcfd162a82c49181a2d480f4b193acdfa6b0e22f4a8448286162630'
@@ -123,11 +125,13 @@ apply_patch_with_msg() {
 
 prepare() {
   [[ -d ${srcdir}/${_pkgfqn} ]] && rm -rf ${srcdir}/${_pkgfqn}
+  msg2 "Extracting source tarball..."
   tar -xJf ${srcdir}/${_pkgfile}.tar.xz -C ${srcdir} --exclude=${_pkgfqn}/{qtandroidextras,qtmacextras,qtwayland,qtwebengine,qtx11extras} || true
 
   cd ${srcdir}/${_pkgfqn}
 
   # MSYS2 gperf cannot handle \r\n.
+  msg2 "Converting line endings of .gperf files..."
   find . -name "*.gperf" -exec dos2unix "{}" \;
 
   apply_patch_with_msg \
@@ -136,6 +140,7 @@ prepare() {
     0002-qt-5.8.0-configure-gcc-before-clang.patch \
     0003-enable-avx-support.patch \
     0004-fix-linking-again-different-static-libs.patch \
+    0005-qtconnectivity-add-callback-attribute.patch \
     0006-qt-5.3.0-win_flex-replace.patch \
     0007-qt-5.3.0-win32-g-Enable-static-builds.patch \
     0008-qt-5.3.0-win32-g-Add-QMAKE_EXTENSION_IMPORTLIB-defaulting-to-.patch \
-- 
2.42.0

