diff -ur old/qtbase/src/angle/src/compiler/translator.pro new/qtbase/src/angle/src/compiler/translator.pro
--- old/qtbase/src/angle/src/compiler/translator.pro	2019-04-16 21:15:23.905543700 +0200
+++ new/qtbase/src/angle/src/compiler/translator.pro	2019-04-16 22:32:24.286196300 +0200
@@ -194,7 +194,7 @@
 
 
 # NOTE: 'flex' and 'bison' can be found in qt5/gnuwin32/bin
-flex.commands = flex --noline --nounistd --outfile=${QMAKE_FILE_OUT}_lex.cpp ${QMAKE_FILE_NAME}
+flex.commands = flex --noline --nounistd --outfile=${QMAKE_FILE_OUT} ${QMAKE_FILE_NAME}
 flex.output = $${BUILDSUBDIR}${QMAKE_FILE_BASE}_lex.cpp
 flex.input = FLEX_SOURCES
 flex.dependency_type = TYPE_C
